Make goal node state change commands systemd aware

bricerisingalgorand opened the pull request:
## Summary

I added a property to libgoal/system.go where we can set whether or not our algod process is managed by systemd. This adds the feature, but without automation for ensuring that the field is set to true on all systemd installs

## Test Plan

I ran it locally on OSX and linux and it appeared to work appropriately. Making a PR to ensure all integration tests are ran

pzbitskiy commented on libgoal/system.go:
```diff
@@ -72,3 +73,16 @@ func AlgorandDataIsPrivate(dataDir string) bool {
+	if err != nil {
+		return false
+	}
+	return sc.SystemdManaged
```
where do we set `SystemdManaged` or `systemd_managed` ?

bricerisingalgorand replied:
In the system.json file of the data dir

algobolson replied:
go ahead and add this to `installer/system.json` which is included in RPM and APT packages which install on systemd distros?

bricerisingalgorand replied:
Cool, I'm adding that with my next commit once I have the unit tests working

bricerisingalgorand commented:
I added tests using expect to try to make sure messages coming from goal on systemd enabled computers look appropriate. This tests that the goal cli is using the field in system.json appropriately. I also added systemd_managed: true as a default value in our system.json file that gets published with linux installs. I looked into wring a full e2e test to make sure that linux installs have that set and it's working appropriately, but our automation for creating linux packages needs a lot of improvement before it can be used conveniently for that purpose. I think we should hold off on writing that test once our automation is in a better state.
